src/backer/cardaning.py

```diff
@@ -59,44 +65,104 @@ def __init__(self, name='backer', hab=None, ks=None):
 
                 # Metadata. accept int key type
                 builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
 
                 signed_tx = builder.build_and_sign([self.payment_signing_key],
                                                 change_address=self.spending_addr,
                                                 merge_change=True)
+                
                 # Submit transaction
-                self.context.submit_tx(signed_tx.to_cbor())  
-                self.pending_kel = bytearray()
-        except Exception as e:    
-            logger.critical(f"Error: {e}")
+                self.context.submit_tx(signed_tx.to_cbor())
+                transId = str(signed_tx.transaction_body.inputs[0].transaction_id)
+                trans = {
+                    "id": transId,
+                    "kel": kel.decode('utf-8'),
+                    "publish_time": help.toIso8601()
+                }
+                self.keldbConfirming.pin(keys=transId, val=json.dumps(trans).encode('utf-8'))
+        except Exception as e:
+            logger.critical(f"Cannot submit transaction: {e}")
+
+    def emptyPendingKEL(self):
+        self.pending_kel = bytearray()
+
+    def flushQueue(self):
+        if self.pending_kel is not None and self.pending_kel != b"":
+            self.submitKelTx(self.pending_kel)        
+            self.pending_kel = bytearray()
+
+    def slotToTime(self, slot):
+        timeStamp = SHELLY_UNIX + slot
+        return datetime.fromtimestamp(timeStamp)
+
+    def getConfirmingTrans(self, transId):
+        return self.keldbConfirming.get(transId)
+
+    def confirmTrans(self):
+        try:
+            for keys, item in self.keldbConfirming.getItemIter():
+                if item is None:
+                    continue
+                
+                item = json.loads(item)
+                (blockSlot, blockHeight) = (item["block_slot"], item["block_height"]) if 'block_slot' in item.keys() else (0, 0)
+                publishTime = datetime.fromisoformat(item["publish_time"])
+                onChainTime = help.toIso8601(self.slotToTime(blockSlot))
+                onChainTime = datetime.fromisoformat(onChainTime).replace(tzinfo=timezone.utc)
+                confirmTime = datetime.fromisoformat(help.toIso8601())
+
+                # Check for confirmation
+                if self.tipHeight - int(blockHeight) >= TRANSACTION_DEEP and (onChainTime - publishTime).total_seconds() <= TRANSACTION_CONFIRMATION_TIMEOUT:
+                    self.keldbConfirming.rem(keys)
+                    continue
+
+                if (confirmTime - publishTime).total_seconds() > TRANSACTION_CONFIRMATION_TIMEOUT:
+                    self.submitKelTx(item['kel'].encode('utf-8'))
+                    self.keldbConfirming.rem(keys)
+        except Exception as e:
+            logger.critical(f"Cannot confirm transaction: {e}")
+
+    def updateTrans(self, transId, blockHeight, blockSlot):
```

Comment:
We can just pass the entire transaction instead of the ID and getting from the DB. We've already tried to fetch it from the DB to call this function. (`getConfirmingTrans`)

That would avoid the if statement.

I think we can remove the try catch as it shouldn't fail. If it fails for DB not being available, a crash and restart is ok